### PR TITLE
1.17 move Sand Scuttler out of scorpion

### PR DESCRIPTION
**Pull request by doofus-01**
The potential unit race help menu tree rearrangements in #7228 reminded me how terrible the "variations" look there (#6011), and it's really only the zombies and the sand scuttler that use the feature, at least for core.  It's not super clear how to deal with the zombies, but the sand scuttler really stands out and can be more easily fixed/removed.

I used `[base_unit]` in case there was some reason to have the Giant Scorpion and Sand Scuttler remain linked.

### CODE REVIEW

**Comment by CelticMinstrel** [comment]
Hmm, this could be a problem. The sand scuttler exists in 1.16 too, doesn't it?

**Comment by doofus-01** [comment]
Yes, unfortunately, so it will be a pain for UMC migration.  Maybe this can be addressed with wmllint?  But I wonder how much "scuttler" is actually used.

**Comment by CelticMinstrel** [comment]
Maybe it can be addressed in wmllint, but I think we could just leave the scuttler variation in place for 1.18 and deprecate it to be removed in 1.19. There is a DEPRECATED_UNIT macro which is meant for deprecating units, but I think it should also work to deprecate a variation if you placed it in the `[variation]` tag. I haven't tested this on a variation, however…

**Comment by doofus-01** [comment]
I disagree with leaving it.  This isn't some "under-the-hood" API change, this is pretty visible and it is bad enough to need fixing, not fading out six years from now.  The scuttler was a band-aid in the first place, and there is a simple fix.  Well, I'm not sure how simple for wmllint, but generating a sensible error message should be possible.

**Review by knyghtmare — APPROVED** [review]

**Comment by CelticMinstrel** [comment]
> This isn't some "under-the-hood" API change

It is a change that will hard break add-ons, which is not good no matter how important the fix is. I don't recall exactly what happens if you try to place a unit with a non-existent variation, but I can't imagine it would be good. It's probably not as bad as a missing unit, I guess. Probably the best thing that could happen is that it reverts to the default variation (and possibly alters the balance of the scenario).

**Comment by CelticMinstrel** [comment]
> It's not super clear how to deal with the zombies

They have a strong reason to stay as variations, because they're the output of a plague special. As long as plague works the way it does, we can't go and change how the zombie unit is structured.

**Comment by knyghtmare** [comment]
> Unless you can prove that "scuttler" is not used anywhere in 1.16 right now, I don't think it's a good idea to skip the deprecation process. Even then it's not really a good idea, but could be justified.

The three use cases of scuttler I have seen is probably Mercenaries, EI-rework and HoD. The first 2 are in 1.17.x and will be able to adjust, I suppose. HoD is still very early pre-alpha so I can easily make the change with little to no conflict.

**Comment by CelticMinstrel** [comment]
Uses in 1.17.x don't matter. If the variation is completely unused (or doesn't exist) in 1.16, then there is basically no problem.

I don't know for certain whether it exists in 1.16, does someone want to confirm that first?

**Comment by doofus-01** [comment]
If we truly couldn't touch this thing because some UMC used it in 1.16, that would mean our deprecation process needed revision.  Fortunately, we don't need to be that sclerotic, the variation can be hidden.

File: data/core/units/monsters/Giant_Scorpion.cfg [hunk 1]
```diff
@@ -106,4 +106,101 @@
         [/frame]
         {SOUND:HIT_AND_MISS spear.ogg {SOUND_LIST:MISS} -100}
     [/attack_anim]
+    [variation]
+        variation_id=scuttler
+        variation_name= _ "scuttler"
+        name= _ "Sand Scuttler"
+        inherit=yes
+        hide_help=yes
+        inherit=yes
```
Comment (knyghtmare) [inline data/core/units/monsters/Giant_Scorpion.cfg:115]: duplicated `inherits=yes`

**Review by knyghtmare — CHANGES_REQUESTED** [review]

File: data/core/units/monsters/Giant_Scorpion.cfg (hunk 1 above)
Comment (doofus-01) [reply data/core/units/monsters/Giant_Scorpion.cfg]: removed

**Review by knyghtmare — APPROVED** [review]

**Pull request merged**